spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -570,7 +573,7 @@ protected String mode() {
 
     @Override
     protected void doCommit(long epochId, WriterCommitMessage[] messages) {
-      AppendFiles append = table.newFastAppend();
+      AppendFiles append = table.newFastAppend().toBranch(branch);
```

Comment:
Are these additional calls to `toBranch` needed, since the `commit` method calls `toBranch`?